automation/check-patch.k8s-1.14.2-kind-sriov.sh

```diff
@@ -0,0 +1,240 @@
+#!/bin/bash -e
+set -x
+
+#############################################################
+# CI Notes TODO Move in a more appropriate location
+# Each CI node labeled as sriov-enable should have the following setup in place:
+# - have the vfio_pci module loaded with a modprobe.d rule
+# - since we use kind, in order to have the dns working we need to
+#   - have the br_netfilter module loaded with a modprobe.d rule
+#   - have the following lines on /etc/sysctl.conf (see https://stackoverflow.com/questions/48148838/kube-dns-error-reply-from-unexpected-source)
+#               net.bridge.bridge-nf-call-ip6tables = 1
+#               net.bridge.bridge-nf-call-iptables = 1
+#               net.bridge.bridge-nf-call-arptables = 1
+#
+#############################################################
+
+
+#############################################################
+# This is based on https://github.com/SchSeba/kubevirt-docker
+#############################################################
+
+export NO_PROXY="localhost,127.0.0.1,172.17.0.2"
+
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="${ARTIFACTS-$WORKSPACE/exported-artifacts}"
+
+CLUSTER_NAME=sriov-ci
+CLUSTER_CONTROL_PLANE=${CLUSTER_NAME}-control-plane
+CONTAINER_REGISTRY_HOST="localhost:5000"
+
+CLUSTER_CMD="docker exec -it -d ${CLUSTER_CONTROL_PLANE}"
+
+KUBEVIRT_PATH=`pwd`
+CLUSTER_DIR="cluster-up/cluster/k8s-1.14.2-kind-sriov"
+MANIFESTS_DIR="${CLUSTER_DIR}/manifests"
+
+
+function wait_containers_ready {
+    echo "Waiting for all containers to become ready ..."
+    kubectl wait --for=condition=Ready pod --all -n kube-system --timeout 12m
+}
+
+function wait_kubevirt_up {
+    echo "Waiting for kubevirt to be ready ..."
+    kubectl wait --for=condition=Ready pod --all -n kubevirt --timeout 12m
+    kubectl wait --for=condition=Ready pod --all -n cdi --timeout 12m
+}
+
+function wait_kind_up {
+    echo "Waiting for kind to be ready ..."  
+    while [ -z "$(docker exec --privileged ${CLUSTER_CONTROL_PLANE} kubectl --kubeconfig=/etc/kubernetes/admin.conf get nodes --selector=node-role.kubernetes.io/master -o=jsonpath='{.items..status.conditions[-1:].status}' | grep True)" ]; do
+        echo "Waiting for kind to be ready ..."        
+	    sleep 10
+    done
+    echo "Waiting for dns to be ready ..."        
+    kubectl wait -n kube-system --timeout=12m --for=condition=Ready -l k8s-app=kube-dns pods
+}
+
+function collect_artifacts {
+    kind export logs ${ARTIFACTS_PATH} --name=${CLUSTER_NAME} || true
+}
+
+function finish {
+    collect_artifacts
+    echo "Printing all pods"
+    kubectl get pods --all-namespaces -o wide
+    kind delete cluster --name=${CLUSTER_NAME}
+}
+
+function enable_vfio {
+    mount -o remount,rw /sys    #need this to move devices to vfio drivers
+    counter=0
+    for file in $(find /sys/devices/ -name *sriov_totalvfs*); do
```

Comment:
is there a particular reason to not call tools/util/vfio.sh?

Reply:
Added this local version mostly because I had an intermediate version that was activating a limited number of vfs.
Guess we could revert to invoking vfio.sh as soon as https://github.com/kubevirt/kubevirt/pull/2444 is merged